packages/server/test/integration/health.test.ts

```diff
@@ -0,0 +1,97 @@
+import { Client } from '../../src/clients/types'
+import { Conduit } from '../../src/conduits/types'
+import { HealthService } from '../../src/health/service'
+import { HealthEventType, HealthReport } from '../../src/health/types'
+import { app, randStr, setupApp } from './utils'
+
+const TEST_CHANNEL = 'telegram'
+
+describe('Health', () => {
+  let health: HealthService
+  let querySpy: jest.SpyInstance
+  let state: { client: Client; conduit: Conduit; eventData?: any; eventData2?: any; clientHealth?: HealthReport }
+
+  beforeAll(async () => {
+    await setupApp()
+    health = app.health
+    querySpy = jest.spyOn(health as any, 'query')
+
+    const provider = await app.providers.create(randStr(), false)
+    const client = await app.clients.create(provider.id, await app.clients.generateToken())
+    const conduit = await app.conduits.create(provider.id, app.channels.getByName(TEST_CHANNEL).meta.id, {
+      botToken: randStr()
+    })
+
+    state = {
+      client,
+      conduit
+    }
+  })
+
+  afterAll(async () => {
+    await app.destroy()
+  })
+
+  beforeEach(async () => {
+    app.caching.resetAll()
+  })
+
+  test('Register health event', async () => {
+    const data = { val: randStr() }
+    await health.register(state.conduit.id, HealthEventType.StartFailure, data)
+
+    state.eventData = data
```

Comment:
Can you simply add something like `expect(data).toBeDefined()`

Reply:
Data is defined earlier. The health.register function doesn't return anything

Reply:
Yeah right, I misread the test, my bad!